nuitka/plugins/standard/standard.nuitka-package.config.yml

```diff
@@ -3823,11 +3823,17 @@
     dirs:
       - '.'
 
-- module-name: 'pygame' # checksum: 5fbb83cf
+- module-name: 'pygame' # checksum: f29246e3
   data-files:
     patterns:
       - 'freesansbold.ttf'
```

Comment:
This isn't related to the DLL issue, but is something else I noticed that would be nice to fix.

Here is the logic we use in the pygame-ce pyinstaller hook for including the pygame icon file, would be nice to have this in here as well, maybe just include anything that matches the `pygame_icon` prefix?

```py
if platform.system() == "Darwin":
    _append_to_datas("pygame_icon_mac.bmp")
else:
    _append_to_datas("pygame_icon.bmp")
```

Reply:
The proper way is to make separate blocks for each platform so just the necessary files are used.

Reply:
Actually the schema outlaws that, but the code will work with it, but I will change that for 2.4 release.

Something like this ought to work:

```
data-files:
    - patterns:
        - 'freesansbold.ttf'
    - patterns:     
         - 'pygame_icon_mac.bmp'
      when: 'macos'
    - patterns:     
         - 'pygame_icon.bmp'
      when: 'not macos'
```

Previously we never had a need for this to be a list, but apparently, we should. I will make a transition for develop right before 2.4 releases, you can already use the code in the PR and igore the checker error.

Reply:
> Actually the schema outlaws that, but the code will work with it, but I will change that for 2.4 release.
> 
> Something like this ought to work:
> 
> ```
> data-files:
>     - patterns:
>         - 'freesansbold.ttf'
>     - patterns:     
>          - 'pygame_icon_mac.bmp'
>       when: 'macos'
>     - patterns:     
>          - 'pygame_icon.bmp'
>       when: 'not macos'
> ```
> 
> Previously we never had a need for this to be a list, but apparently, we should. I will make a transition for develop right before 2.4 releases, you can already use the code in the PR and igore the checker error.

I might be misunderstanding what you're saying. Using this version, the commit hook yells at me that it's not valid, so it won't even let me commit. If I remove the `-` from before `patterns`, it then spits out a duplicate key error because `patterns` is repeated

Reply:
Here's what I mean:
```yaml
- module-name: 'pygame' # checksum: f29246e3
  data-files:
    - patterns:
      - 'freesansbold.ttf'
    - patterns:
      - 'pygame_icon_mac.bmp'
      when: 'macos'
    - patterns:
      - 'pygame_icon.bmp'
      when: 'not macos'

  dlls:
    - from_filenames:
        relative_path: '.'
        prefixes:
          - ''

  options:
    checks:
      - description: 'PyGame is a GUI framework'
        macos_bundle: 'recommend'
```
yields this error 
```
Traceback (most recent call last):
  File "C:\Users\andre\Projects\Nuitka\bin\autoformat-nuitka-source", line 20, in <module>
    main()
  File "C:\Users\andre\Projects\Nuitka\nuitka\tools\quality\auto_format\__main__.py", line 105, in main
    autoFormatFile(desc["src_path"], git_stage=desc)
  File "C:\Users\andre\Projects\Nuitka\nuitka\tools\quality\auto_format\AutoFormat.py", line 798, in autoFormatFile
    formatYaml(tmp_filename, ignore_diff=ignore_yaml_diff)
  File "C:\Users\andre\Projects\Nuitka\nuitka\tools\quality\auto_format\YamlFormatter.py", line 313, in formatYaml
    sorted_entry["data-files"] = _reorderDictionary(
                                 ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\andre\Projects\Nuitka\nuitka\tools\quality\auto_format\YamlFormatter.py", line 159, in _reorderDictionary
    entry._items(),
    ^^^^^^^^^^^^
AttributeError: 'CommentedSeq' object has no attribute '_items'
```